src/GraphRequest.ts

```diff
@@ -199,38 +199,40 @@ export class GraphRequest {
 
     delete(callback?:GraphRequestCallback):Promise<any> {
         let url = this.buildFullUrl();
-        return this.sendRequestAndRouteResponse(request.del(url), callback)
+        return this.sendRequestAndRouteResponse(
+            new Request(url, { method: RequestMethod.DELETE, headers: new Headers() }),
+            callback
+        );
     }
 
     patch(content:any, callback?:GraphRequestCallback):Promise<any> {
         let url = this.buildFullUrl();
         
         return this.sendRequestAndRouteResponse(
-            request
-                .patch(url)
-                .send(content),
+            new Request(url, { method: RequestMethod.PATCH, body: content, headers: new Headers() }),
             callback
         );
     }
 
     post(content:any, callback?:GraphRequestCallback):Promise<any> {
         let url = this.buildFullUrl();
         return this.sendRequestAndRouteResponse(
-            request
-                .post(url)
-                .send(content),
-                callback
+            new Request(url, { method: RequestMethod.POST, body: content, headers: new Headers() }),
+            callback
         );
     }
 
     put(content:any, callback?:GraphRequestCallback):Promise<any> {
         let url = this.buildFullUrl();
         return this.sendRequestAndRouteResponse(
-            request
-                .put(url)
-                .type('application/octet-stream')
-                .send(content),
-                callback
+            new Request(
+                url,
+                { 
+                    method: RequestMethod.PUT,
+                    body: content,
+                    headers: new Headers({ 'Content-Type' : 'application/octet-stream' })
```

Comment:
@MIchaelMainer wouldn't this be a breaking change? Isn't the original code setting this header?

Reply:
Shoot. Yes, you're right. This would be breaking. Unfortunately, this implementation is wrong. We can Put other content types in other calls plus, there is a PutStream function that also hard codes this content type. For now, we should leave this broke (not happy about this).

Example
https://developer.microsoft.com/en-us/graph/docs/api-reference/beta/api/team_put_teams

Reply:
@lahuey Can you make it so that when the customer sets the content-type header, it checks whether it exists and it removes the existing one if it does. That way, we won't break existing customers who have taken a dependency on this behavior.

Reply:
changed this.configureRequest to use Headers.set instead of Headers.append